src/Context/dataContext.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useContext } from "react";
+import { useEffect, useReducer } from "react";
+import { DataReducer } from "../Reducer/video-reducer";
+import { Abbreviations } from "../services/abbreviations";
+import { getCategoryService, getVideoService } from "../services/dataservice";
+import { useFetch } from "../custom-hooks/useFetch";
+const VideoProvider = ({ children }) => {
+  const { data: video } = useFetch("/api/videos", "videos");
+  const DatainitialState = {
+    videos: [],
+    categories: [],
+    sortBy: "All",
+  };
+  const [Datastate, Datadispatch] = useReducer(DataReducer, DatainitialState);
+  useEffect(() => {
+    // (async () => {
+    //   try {
+    //     const vidData = await getVideoService();
+    //     Datadispatch({
+    //       type: Abbreviations.INITIAL_VIDEOS,
+    //       payload: vidData.data.videos,
+    //     });
+    //   } catch (error) {
+    //     console.log(error);
+    //   }
+    // })();
```

Comment:
remove this if not needed

Reply:
I will do so finally, as it branch, so trying stuffs.